### PR TITLE
DOP-4818: Not Found page - Dark Mode

### PR DESCRIPTION
**Pull request by mmeigs**
### Stories/Links:

DOP-4818

### Current Behavior:

[404](https://www.mongodb.com/docs/404/)

### Staging Links:

[404](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/add-title/docs-404/matt.meigs/DOP-4818-darkmode-404/index.html)

### Notes:

Changes link styling to dynamic dark-mode supported.

Cleans up some styles in the NotFound page.

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

**Review by mayaraman19 — COMMENTED** [review]
Looks good! One tiny comment is that when initially rendering in dark mode, the contact support link (re-)renders 3 times. Wondering there's a way to change that behavior so it only renders like 1-2 times?

**Comment by mmeigs** [comment]
> Looks good! One tiny comment is that when initially rendering in dark mode, the contact support link (re-)renders 3 times. Wondering there's a way to change that behavior so it only renders like 1-2 times?

Sure! I made a change that seems to help that! It was because of a battle between all the styling choices we and LG have going on: styled-components, emotion, and then dark mode hydration.

File: src/templates/NotFound.js [hunk 1]
```diff
@@ -54,79 +51,80 @@ const NotFoundImage = () => {
...
+        </Link>
+      </LinkContainer>
     </ErrorBox>
   );
 };
 
+const NotFoundContainer = styled.div`
+  align-items: center;
+  display: flex;
+  flex-flow: row-reverse wrap;
```
Comment (seungpark) [inline src/templates/NotFound.js:111]: [nit] is there a reason we do `` then `<ErrorBoxContainer/>` in that order, and then reverse?

File: src/templates/NotFound.js [hunk 2]
```diff
@@ -54,79 +51,80 @@ const NotFoundImage = () => {
   );
 };
 
+const ErrorTitle = styled.p`
```
Comment (seungpark) [inline src/templates/NotFound.js:54]: [minor nit] wondering why we are also not using LG/typography here

File: src/templates/NotFound.js (hunk 1 above)
Comment (mmeigs) [reply src/templates/NotFound.js:111]: I didn't change any html structure in this PR, but I'm open to doing so if you'd like. 
I'm just not sure what you mean? Are you speaking of the html or the naming?

File: src/templates/NotFound.js (hunk 2 above)
Comment (mmeigs) [reply src/templates/NotFound.js:54]: I went for least-invasive. But I'm good with changing this over!

File: src/templates/NotFound.js (hunk 1 above)
Comment (seungpark) [reply src/templates/NotFound.js:111]: im speaking of how the DOM is generally
```
      <NotFoundContainer>
        <NotFoundImage />
        <ErrorBoxContainer />
      </NotFoundContainer>
```

but they are shown in reverse (?) not a huge deal as i see this was just simply moved from css prop to classname

File: src/components/Banner/SiteBanner.js [hunk 3]
```diff
@@ -56,7 +60,11 @@ const SiteBanner = () => {
 
   return (
     <StyledBannerContainer href={bannerContent.url} title={bannerContent.altText}>
-      <StyledBannerContent imgPath={bannerContent.imgPath} mobileImgPath={bannerContent.mobileImgPath} />
+      <StyledBannerContent
+        imgPath={bannerContent.imgPath}
+        tabletImgPath={bannerContent.tabletImgPath}
+        mobileImgPath={bannerContent.mobileImgPath}
+      />
```
Comment (rayangler) [inline src/components/Banner/SiteBanner.js:67]: Just making sure: are we intentionally adding this site banner change to this PR?

**Comment by mmeigs** [comment]
> generally LGTM! can you add the figma or design page that this matches up to? thank you!

There was no Figma! I asked [Amelia in Slack](https://mongodb.slack.com/archives/C076Z47AFTL/p1721227102724589) about it and she said just the Contact Support change.

I've also re-staged this in the link above so you can see with changes in.

File: src/templates/NotFound.js [hunk 4]
```diff
@@ -54,79 +52,85 @@ const NotFoundImage = () => {
   );
 };
 
+const ErrorTitle = styled(Body)`
```
Comment (rayangler) [inline src/templates/NotFound.js:55]: We should avoid wrapping LG components in `styled` and try to use LG's emotion package. The 2 different styling methods can sometimes cause a delay where the original LG styling shows first before showing the overwritten styling

File: src/components/Banner/SiteBanner.js (hunk 3 above)
Comment (mmeigs) [reply src/components/Banner/SiteBanner.js]: Yes, I truly just SCREAMED a second ago when I saw I brought over changes from another branch. it's gone now, sorry!

File: src/templates/NotFound.js [hunk 5]
```diff
@@ -54,79 +52,87 @@ const NotFoundImage = () => {
...
+  line-height: 40px;
+
+  @media ${theme.screenSize.upToSmall} {
+    font-size: ${theme.fontSize.h2};
+  }
+`;
+
+const errorSubtitleStyling = css`
+  ${commonBodyStyling}
+  font-size: ${theme.fontSize.default};
```
Comment (seungpark) [inline src/templates/NotFound.js:73]: [minor] can remove. seems like we are overwriting default body styles.

**Review by seungpark — APPROVED** [review]
left some minor comments on typography but looks great overall, thanks @mmeigs !

**Review by rayangler — APPROVED** [review]

**Pull request merged**